### PR TITLE
Add the remaining unit tests and fix them

### PR DESCRIPTION
**Pull request by kccqzy**
This test probably also doesn't work like the one added before. I implemented a bunch of inmemory tree stuff to get this compiled. The stuff was from
https://github.com/AlCutter/trillian/blob/master/internal/merkle/inmemory/merkle_tree.go and
https://github.com/AlCutter/trillian/blob/master/merkle/logverifier/log_verifier_test.go

### CODE REVIEW

File: libjcat/jcat-self-test.c [hunk 1]
```diff
@@ -2613,9 +2613,308 @@ TestVerifyInclusionProof(void)
...
+	}
+	return subtree_root;
+}
+
+static GByteArray *
+inmemoryTreeRootAtSnapshot(inmemoryTree *tree, gint64 snapshot)
+{
+	if (snapshot == 0) {
+		return g_bytes_unref_to_array(sha256EmptyTreeHash());
+	}
```
Comment (hughsie) [inline libjcat/jcat-self-test.c:2818]: just stylisticly, you don't need {} for single line statements like this.

File: libjcat/jcat-self-test.c [hunk 2]
```diff
@@ -2613,9 +2613,308 @@ TestVerifyInclusionProof(void)
...
+{
+	GPtrArray *path = g_ptr_array_new_with_free_func((GDestroyNotify)g_byte_array_unref);
+	gint64 last_node = (snapshot - 1) >> level;
+	if (snapshot == 0 || level >= tree->levelCount || node > last_node ||
+	    snapshot > inmemoryTreeLeafCount(tree))
+		return path;
+	if (snapshot > tree->leavesProcessed) {
+		inmemoryTreeUpdateToSnapshot(tree, snapshot);
+	}
+	while (last_node) {
```
Comment (hughsie) [inline libjcat/jcat-self-test.c:2848]: maybe add != 0 to make this clear it's an int.

File: libjcat/jcat-self-test.c (hunk 1 above)
Comment (kccqzy) [reply libjcat/jcat-self-test.c]: Done.

File: libjcat/jcat-self-test.c (hunk 2 above)
Comment (kccqzy) [reply libjcat/jcat-self-test.c]: Done.

**Comment by kccqzy** [comment]
All the unit tests now pass.

**Review by hughsie — APPROVED** [review]
woah, that looked like tons of work, good job!

**Pull request merged**